uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveManagerTest.php

```diff
@@ -25,109 +29,109 @@ public function tearDown() {
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
 
     $today = new DateTime('today');
     $tomorrow = new DateTime('tomorrow');
-    $this->setContactAsLeaveApproverOf($manager, $staffMember, $today->format('Y-m-d'), $tomorrow->format('Y-m-d'));
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, $today->format('Y-m-d'), $tomorrow->format('Y-m-d'));
 
-    $this->assertTrue($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertTrue($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
   }
 
   public function testIsContactManagedByWhenTheresNoActiveRelationshipForTheCurrentDate() {
-    $manager = ContactFabricator::fabricate();
-    $staffMember = ContactFabricator::fabricate();
-
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
 
     // Set a relationship in the past
     $startDate = new DateTime('-10 days');
     $endDate = new DateTime('-1 day');
-    $this->setContactAsLeaveApproverOf($manager, $staffMember, $startDate->format('Y-m-d'), $endDate->format('Y-m-d'));
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, $startDate->format('Y-m-d'), $endDate->format('Y-m-d'));
 
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
 
     // Set a relationship in the future
     $startDate = new DateTime('+1 day');
-    $this->setContactAsLeaveApproverOf($manager, $staffMember, $startDate->format('Y-m-d'));
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, $startDate->format('Y-m-d'));
 
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
   }
 
   public function testIsContactManagedByWhenTheCurrentRelationshipIsNotActive() {
-    $manager = ContactFabricator::fabricate();
-    $staffMember = ContactFabricator::fabricate();
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
+
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, null, null, false);
+
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
+  }
 
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+  public function testIsContactManagedByWhenCurrentUserHasMoreThanOneRelationshipWithTheEmployee() {
```

Comment:
It's possible to have more than one active relationship at the same time. I added a new test to cover this scenario